### DISCUSSION TITLE
Access json data using a variable name as lookup value

### DISCUSSION THREAD

**Discussion by alexanderameye** [11ty/eleventy#1797] [Q&A] [answered]
`Like` in the eleventy base template I have a list of tags that I display like this.

```html
<ul class="note-tags">
    {%- for tag in tags | filterTagList -%}
        {%- set color = tagcolors.Math -%}
        <li>
            <a style="background-color: {{color}};" href="/tags/{{tag}}">{{tag}}</a>
        </li>
    {%- endfor -%}
</ul>
```

I have a json data file like this that is supposed to allow me to give each tag a color.

`{
    "Math": "#E2D4FF",
    "Graphics": "#FFDE69"
}`

Now I'm trying to do this where I access the right color from the json data based on the tag but this syntax is not right and I get an error "expected name as lookup value, got {"

```html
<ul class="note-tags">
    {%- for tag in tags | filterTagList -%}
        {%- set color = tagcolors.{{tag}}} -%}
        <li>
            <a style="background-color: {{color}};" href="/tags/{{tag}}">{{tag}}</a>
        </li>
    {%- endfor -%}
</ul>
```

Is there a suggested workaround for this? I would love to be able to define the colors for a tag in one spot.

Thanks in advance!

**Comment by pdehaan** [comment, marked as answer]
I think you want:

```njk
{%- set color = tagcolors[tag] -%}
```

**Reply by alexanderameye** [reply]
I just saw that in the issue below mine where you commented!

https://github.com/11ty/eleventy/discussions/1796#discussioncomment-790757

Thanks a bunch :)
